src/wp-includes/class-wp-html-attribute-token.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * HTML Tag Processor: Attribute token structure class.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ * @since 6.2.0
+ */
+
+if ( ! class_exists( 'WP_HTML_Attribute_Token' ) ) :
+
+/**
+ * Data structure for the attribute token that allows to drastically improve performance.
+ *
+ * This class is for internal usage of the WP_HTML_Tag_Processor class.
+ *
+ * @access private
+ * @since 6.2.0
+ *
+ * @see WP_HTML_Tag_Processor
+ */
+class WP_HTML_Attribute_Token {
```

Comment:
@costdev is there a benefit here to `final`? I don't mind much, and have finalized them in 8b507e5417, but I didn't think it was too important to prevent changes to them (I'm generally more liberal in allowing people to do what they want with their sites).

Reply:
> I see your feedback on the original PR asked to lock this code down

@costdev this was "while in development". Some of the security features weren't fully implemented yet, and there were some questions on expected use, limitations, etc. All of these have been resolved since.

Looking at the expected usage `final` may be appropriate, perhaps, but will defer to the authors. Generally WP uses actions and filters to extent functionality. Extending with sub-classes may be harder to manage when more than one plugin is doing it. On the other hand `final` may prevent Gutenberg from updating some of the functionality, perhaps better to leave it off.

Reply:
If we make the class `final` and with no actions/filters in place right now(unless I missed something), it will be really difficult to extend some functionality in these classes in GB, no?

Reply:
@dmsnell @azaozz @ntsekouras Thanks for your responses! Yes, let's not make any of these classes `final` so that we don't make things difficult for work in GB.

Reply:
reverted to the original non-final versions in 2d1411a8ea